js/src/components/check/CheckList.tsx

```diff
@@ -0,0 +1,172 @@
+import "react-data-grid/lib/styles.css";
+import React, { useEffect, useState } from "react";
+import { Check, reorderChecks, updateCheck } from "@/lib/api/checks";
+import { Box, Flex, Icon, VStack } from "@chakra-ui/react";
+import { cacheKeys } from "@/lib/api/cacheKeys";
+import { useMutation, useQueryClient } from "@tanstack/react-query";
+import _ from "lodash";
+import { FaCheckCircle } from "react-icons/fa";
+import {
+  TbChecklist,
+  TbSql,
+  TbSchema,
+  TbAlignBoxLeftStretch,
+  TbChartHistogram,
+} from "react-icons/tb";
+import { IconType } from "react-icons";
+import { FiAlignLeft } from "react-icons/fi";
+import {
+  DragDropContext,
+  Draggable,
+  DropResult,
+  Droppable,
+} from "@hello-pangea/dnd";
+
+const ChecklistItem = ({
+  check,
+  selected,
+  onSelect,
+}: {
+  check: Check;
+  selected: boolean;
+  onSelect: (checkId: string) => void;
+}) => {
+  const queryClient = useQueryClient();
+  const checkId = check.check_id!;
+  const { mutate } = useMutation({
+    mutationFn: (check: Partial<Check>) => updateCheck(checkId, check),
+    onSuccess: () => {
+      queryClient.invalidateQueries({ queryKey: cacheKeys.check(checkId) });
+      queryClient.invalidateQueries({ queryKey: cacheKeys.checks() });
+    },
+  });
+
+  const handleChange: React.ChangeEventHandler = (event) => {
+    const isChecked: boolean = (event.target as any).checked;
+    mutate({ is_checked: isChecked });
+  };
+
+  const icon: IconType = ((type) => {
+    switch (type) {
+      case "schema_diff":
+        return TbSchema;
+      case "query":
+      case "query_diff":
+        return TbSql;
+      case "value_diff":
+        return TbAlignBoxLeftStretch;
+      case "profile_diff":
+        return TbChartHistogram;
+      case "row_count_diff":
+        return FiAlignLeft;
+      default:
+        return TbChecklist;
+    }
+  })(check.type);
+
+  return (
+    <Flex
+      width="100%"
+      p="10px 20px"
+      cursor="pointer"
+      _hover={{ bg: "gray.200" }}
+      bg={selected ? "gray.100" : "inherit"}
+      onClick={() => onSelect(check.check_id)}
+      alignItems="center"
+      gap="5px"
+    >
+      <Icon as={icon} />
+      <Box
+        flex="1"
+        textOverflow="ellipsis"
+        whiteSpace="nowrap"
+        overflow="hidden"
+      >
+        {check.name}
+      </Box>
+
+      {check.is_checked && <Icon color="green" as={FaCheckCircle} />}
+    </Flex>
+  );
+};
+
+export const CheckList = ({
+  checks,
+  selectedItem,
+  onSelectHandler,
+}: {
+  checks: Check[];
+  selectedItem?: string;
+  onSelectHandler: (checkId: string) => void;
+}) => {
+  const queryClient = useQueryClient();
+  const [orderedChecks, setOrderedChecks] = useState<Check[]>(checks);
```

Comment:
It's ok to move the CheckList to a separate component. However, the `checks` if from parent, and the reordering with orderCheck is in the `CheckList` itself.

Recommended implementation.

Add a `onReorder?: (src, desc) => void` props in `CheckList` component. Implement the reodering logic in parent.

See the similar example here
https://github.com/adazzle/react-data-grid/blob/main/website/demos/RowsReordering.tsx#L62-L72